src/runtime_src/core/tools/xbmgmt2/SubCmdDump.cpp

```diff
@@ -0,0 +1,181 @@
+  commonOptions.add_options()
+    ("device,d", boost::program_options::value<decltype(devices)>(&devices)->multitoken(), "The Bus:Device.Function (e.g., 0000:d8:00.0) device of interest.")
+    ("config,c", boost::program_options::bool_switch(&config), "Dumps the output of system configuration.")
+    ("flash,f", boost::program_options::bool_switch(&flash), "Dumps the output of programmed system image.")
+    ("output,o", boost::program_options::value<decltype(output)>(&output), "Direct the output to the given file")
+    ("help,h", boost::program_options::bool_switch(&help), "Help to use this sub-command")
+  ;
+
+  po::options_description hiddenOptions("Hidden Options");
+
+  po::options_description allOptions("All Options");
+  allOptions.add(commonOptions);
+  allOptions.add(hiddenOptions);
+
+  // Parse sub-command ...
+  po::variables_map vm;
+
+  try {
+    po::store(po::command_line_parser(_options).options(allOptions).run(), vm);
+    po::notify(vm); // Can throw
+  } catch (po::error& e) {
+    std::cerr << "ERROR: " << e.what() << std::endl << std::endl;
+    printHelp(commonOptions, hiddenOptions);
+    return;
+  }
+
+  // Check to see if help was requested or no command was found
+  if (help == true)  {
+    printHelp(commonOptions, hiddenOptions);
+    return;
+  }
+
+  // -- Now process the subcommand --------------------------------------------
+  XBU::verbose("SubCmd: Dump");
+
+  // -- process "device" option -----------------------------------------------
+  XBU::verbose("Option: device");
+  for (auto & str : devices)
+    XBU::verbose(std::string(" ") + str);
+
+  if(devices.empty()) {
+    std::cerr << "ERROR: Please specify a single device using --device option" << "\n\n";
+    printHelp(commonOptions, hiddenOptions);
+    return;
+  }
+
+  // Collect all of the devices of interest
+  std::set<std::string> deviceNames;
+  xrt_core::device_collection deviceCollection;  // The collection of devices to examine
+  for (const auto & deviceName : devices) 
+    deviceNames.insert(boost::algorithm::to_lower_copy(deviceName));
+
+  try {
+    XBU::collect_devices(deviceNames, false /*inUserDomain*/, deviceCollection);
+  } catch (const std::runtime_error& e) {
+    // Catch only the exceptions that we have generated earlier
+    std::cerr << boost::format("ERROR: %s\n") % e.what();
+    return;
+  }
+
+  // enforce 1 device specification
+  if(deviceCollection.size() != 1) {
+    std::cerr << "ERROR: Please specify a single device. Multiple devices are not supported" << "\n\n";
+    printHelp(commonOptions, hiddenOptions);
+    return;
+  }
+
+  // -- process "output" option -----------------------------------------------
+  // Output file
+  XBU::verbose("Option: output: " + output);
+  
+  if (output.empty()) {
+    std::cerr << "ERROR: Please specify an output file using --output option" << "\n\n";
+    printHelp(commonOptions, hiddenOptions);
+    return;
+  }
+  if (!output.empty() && boost::filesystem::exists(output)) {

```

Comment:
Since --output option by itself doesn't specify it if is a directory or file (yes, the help indicates a file).  You **_might_** wish to add a check for a directory, and if given, produce a message indicating that directories are not support. 

This doesn't need to be done with this pull request, but is a nice to have.